src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,127 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float LOAD_FACTOR = 0.75f;
+    Node<K, V>[] table;
+    int threshold;
+    private int size;
+
+    public MyHashMap() {
+    }
 
     @Override
     public void put(K key, V value) {
-
+        putValue(hash(key), key, value);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (table != null) {
+            Node<K, V> head;
+            Node<K, V> returnValue = null;
+            Node<K, V>[] tab = table;
+            head = tab[Math.abs(hash(key) % tab.length)];
+            while (head != null) {
+                if ((key == head.key && key == null) || (key != null && key.equals(head.key))) {
+                    returnValue = head;
+                    break;
+                }
+                head = head.next;
+            }
+            return returnValue.value;
+        } else {
+            return null;
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putValue(int hash, K key, V value) {
+        Node<K, V>[] tab = table;
+        int elementIndex;
+        if (tab == null || tab.length == 0) {
+            tab = resize();
+        }
+        elementIndex = hash(key) % tab.length;
+        if (tab[Math.abs(elementIndex)] == null) {
+            tab[Math.abs(elementIndex)] = new Node<>(hash, key, value, null);
+            size++;
+        } else {
+            Node<K, V> head = tab[Math.abs(elementIndex)];
+            while (head != null) {
+                if ((key == head.key && head.hash == hash && key == null)
+                        || (head.key != null && head.key.equals(key)) && head.hash == hash) {
+                    head.value = value;
+                    break;
+                } else {
+                    if (head.next == null) {
+                        head.next = new Node<>(hash, key, value, null);
+                        size++;
+                        break;
```

Comment:
u don't need to check head.hash for equals, and you also can use method Objects.equals(). And you're using break in if, so you can delete your else statement, and left only 2 if();